DFE-Digital/early-careers-framework
# Push NPQ accept/reject stats to BigQuery

### Context

- We need to be able to reporting for the accept/reject on NPQ validations

### Changes proposed in this pull request

- Add the ability to push data into BigQuery in a streaming append only fashion
- On `NPQValidationData` creation and updating of status stream this data to BigQuery asynchronously

### Guidance to review

- I'm aware of the existence of https://github.com/DFE-Digital/early-careers-framework/pull/941, however it has not been merged yet

### Testing

- Test in review app

### Review Checks

### How can I view this in a review app?

- `cf ssh ecf-review-pr-959`
- start a rails console
- create an application with:
```
record = NPQValidationData.create!(user: User.all.sample, npq_lead_provider: NPQLeadProvider.all.sample, npq_course: NPQCourse.all.sample)
```
- should see `pending` row added to `enrollments_deployed_development` in BigQuery
- update status with:
```
record.update(lead_provider_approval_status: "rejected")
```
- should now see additional row with status `rejected`
